File: design-probes/stateful-index/flows/ingest/flow.py
#!/usr/bin/env python3

# DESIGN PROBE ONLY: hypothetical SDK, complete pseudocode behavior.
from flowmd_sdk import serve_run


async def ingest(run):
    record = await run.effects.call(
        operation_id="upsert-document",
        slot="index",
        method="upsert",
        input=run.input,
    )
    return {"outcome": "done", "output": record}


serve_run(ingest)

File: design-probes/gui-document-desk/flows/search/flow.ts
#!/usr/bin/env bun

// DESIGN PROBE ONLY: hypothetical SDK, complete pseudocode behavior.
import { type JsonValue, serveRun } from "@flowmd/sdk";

interface SearchInput {
  readonly query: string;
  readonly limit: number;
}

serveRun<SearchInput>(async run => {
  const result = await run.effects.call<JsonValue>({
    operationId: "search-index",
    slot: "index",
    method: "search",
    input: run.input as unknown as JsonValue,
  });
  return { outcome: "done", output: result };
});
